cpp/src/arrow/types/primitive-test.cc

```diff
@@ -214,10 +214,10 @@ void TestPrimitiveBuilder<PBoolean>::Check(
     ex_null_bitmap = nullptr;
   }
 
-  auto expected = std::make_shared<BooleanArray>(size, ex_data, ex_null_count,
-      ex_null_bitmap);
-  std::shared_ptr<BooleanArray> result = std::dynamic_pointer_cast<BooleanArray>(
-      builder->Finish());
+  auto expected =
+      std::make_shared<BooleanArray>(size, ex_data, ex_null_count, ex_null_bitmap);
```

Comment:
I'm fine with bin-packing if it means what I think it means (minimizing the number of line breaks while keeping things under 90 chars). I was half doing that by hand already! So feel free to turn it on.